File: JOBCOL BackEnd/backend/src/main/java/com/jobcol/backend/NotificationService/service/NotificationService.java
package com.jobcol.backend.NotificationService.service;

import java.util.List;
import java.util.Optional;

import com.jobcol.backend.NotificationService.model.NotificationType;
import com.jobcol.backend.shared.dto.NotificationDTO;

public interface NotificationService {

    NotificationDTO createNotification(
            Long userId,
            String title,
            String message,
            NotificationType type
    );

    Optional<NotificationDTO> getById(Long id);

    List<NotificationDTO> getUserNotifications(Long userId);

    List<NotificationDTO> getUnreadNotifications(Long userId);

    Long countUnread(Long userId);

    NotificationDTO markAsRead(Long id);

    void markAllAsRead(Long userId);

    void deleteNotification(Long id);

    void createNotificationsBatch(List<Long> userIds,String title,String message,NotificationType type);
    
    void sendEmail(String to, String subject, String body);
    
    NotificationDTO createNotificationOnly(
            Long userId,
            String title,
            String message,
            NotificationType type
    );
} 

File: JOBCOL FrontEnd/src/services/notificationService.js
import api from "./api";

export const getUserNotifications = async (userId) => {
    const response = await api.get(`/notifications/user/${userId}`);
    return response.data;
};

export const getUnreadNotifications = async (userId) => {
    const response = await api.get(`/notifications/user/${userId}/unread`);
    return response.data;
};

export const countUnread = async (userId) => {
    const response = await api.get(`/notifications/user/${userId}/count-unread`);
    return response.data;
};

export const markAsRead = async (notificationId) => {
    const response = await api.put(`/notifications/${notificationId}/read`);
    return response.data;
};

export const markAllAsRead = async (userId) => {
    const response = await api.put(`/notifications/user/${userId}/read-all`);
    return response.data;
};

export const createNotification = async (userId, title, message, type) => {
    const response = await api.post(
        `/notifications/create?userId=${userId}&title=${encodeURIComponent(title)}&message=${encodeURIComponent(message)}&type=${type}`
    );
    return response.data;
};
